lib/mainloop-threaded-worker.h

```diff
@@ -37,12 +37,17 @@ struct _MainLoopThreadedWorker
 
   gpointer data;
   MainLoopWorkerType worker_type;
+  GMutex lock;
+  GCond started_up;
+  gboolean startup_finished;
+  gboolean startup_result;
```

Comment:
Can we somehow group these fields together to show the bool+cond+mutex is there for startup synchronization?
If that's inconvenient, I would just rename `lock` to `startup_lock` to indicate this is for the startup condition, not for protecting the whole class's resources.

Reply:
What if I was using the same lock for exit synchronization? I wouldn't dedicate a separate lock for that. I could rename started_up to startup_cond or even do this:

```
struct
{
  GCond cond;
  gboolean finished;
  gboolean result;
} startup;
```

But I don't think "lock" belongs there. Lock granularity comes up for locks & data structures that are used at a high frequency but in this case a single lock for the current/possibly future uses in this data structure should suffice. I agree
that documenting the locking strategy is a good idea, but taking it into a direction where _all_ synchronization needs to use a separate, specific lock is not right.

I'll implement the struct then you can respond and we discuss :)

Reply:
My idea was based on Rust mutexes, where a lock encapsulates the protected resource/data and only allows access to those resources through RAII guards. This explicitly binds the protected entity to the lock, but I agree, it would be overkill here.

Thank you for talking this through with me.